### ISSUE TITLE
[BUG][BENCHMARK] CUPTI profiler misattributes PyTorch elementwise baseline latencies

### ISSUE THREAD

**Issue by lcy-seso** [tile-ai/TileOPs#543]
### Symptom / Motivation

When benchmarking TileOPs elementwise kernels against PyTorch baselines using the CUPTI profiler backend, **PyTorch baseline latencies are reported as near-zero**, making performance comparisons invalid.

This was discovered during PR #538 (vectorize bool mask loads in where/masked_fill) when reviewer zhen8838 questioned the profiler backend change.

### Root Cause Analysis

tilelang's `_bench_with_cupti` (`tilelang/profiler/bench.py:198`) excludes all CUDA kernels whose name contains the substring `"at::native::vectorized_elementwise"`:

```python
excluded_kernels = "at::native::vectorized_elementwise"
for event in profiler.key_averages():
    total_cuda_time += event.self_device_time_total
    if excluded_kernels in event.key:
        excluded_time += event.self_device_time_total
```

**Intent**: strip `cache.zero_()` overhead inserted between benchmark iterations.

**Reality**: `cache.zero_()` produces a kernel named `vectorized_elementwise_kernel<FillFunctor>`, which belongs to the **same kernel family** that PyTorch uses for all standard elementwise ops. Verified via `torch.profiler`:

| Operation | Kernel name | Matched by filter |
|---|---|---|
| `cache.zero_()` | `vectorized_elementwise_kernel<FillFunctor<int>>` | **YES** (intended) |
| `torch.add(x, y)` | `vectorized_elementwise_kernel<CUDAFunctor_add<float>>` | **YES** (unintended) |
| `torch.where(c, x, y)` | `vectorized_elementwise_kernel<where_kernel_impl...>` | **YES** (unintended) |
| `torch.maximum(x, y)` | `vectorized_elementwise_kernel<maximum_kernel...>` | **YES** (unintended) |
| `torch.relu(x)` | `vectorized_elementwise_kernel<launch_clamp_scalar...>` | **YES** (unintended) |

The overly broad substring match strips **all** PyTorch elementwise kernel time, not just the cache-clearing kernel.

A second independent issue also exists: CUPTI is a **global singleton** — when another process on the machine holds the CUPTI handle (e.g. someone running `ncu`), `do_bench` returns 0. The existing `latency <= 0` fallback to the event backend handles this case.

### Related Files

- `benchmarks/benchmark.py` — TileOPs benchmark base class (`BenchmarkBase.profile()`)
- `tilelang/profiler/bench.py:175-206` — upstream `_bench_with_cupti` (cannot modify directly)
- All `benchmarks/ops/bench_*elementwise*.py` — affected benchmarks

## Goal

Ensure all benchmarks (including elementwise) get accurate CUPTI measurements while preserving the CUPTI singleton fallback mechanism.

## Plan

1. Monkey-patch `tilelang.profiler.bench._bench_with_cupti` at module load time in `benchmarks/benchmark.py` to narrow the exclusion filter from `"at::native::vectorized_elementwise"` to kernels containing **both** `"vectorized_elementwise"` **and** `"FillFunctor"`
2. Add a defensive `hasattr` guard so the patch degrades gracefully if tilelang renames its internal API
3. Preserve the existing CUPTI → event fallback for global singleton contention (ncu usage)
4. Improve event fallback to use `return_mode='median'` for robustness at microsecond latencies
5. File an upstream issue/PR against tilelang to fix the filter at the source (future work)

## Constraints
- Cannot modify tilelang source directly (upstream dependency)
- Must not break non-elementwise benchmarks that rely on CUPTI

## Acceptance Criteria
- [ ] Modified files pass unit tests
- [ ] Elementwise benchmark baseline latencies are non-zero and plausible when using CUPTI backend
- [ ] Non-elementwise benchmarks (gemm, attention, norm, etc.) remain unaffected
- [ ] Patch degrades gracefully (warning, no crash) if tilelang API changes
